std/random.d

```diff
@@ -1934,6 +1934,177 @@ A singleton instance of the default random number generator
     assert(rnd.take(3).sum > 0);
 }
 
+// OS RND code is based on mir.random, https://github.com/libmir/mir-random/blob/master/source/mir/random/engine/package.d
+
+version (FreeBSD)
+    version = UseFileRandom;
+version (linux)
+    version = UseFileRandom;
+
+version (Windows)
+{
+    import core.sys.windows.winbase, core.sys.windows.wincrypt,
+        core.sys.windows.windows, core.sys.windows.winerror,
+        std.windows.syserror;
+
+    struct WindowsRandom
+    {
+    private:
+        ULONG_PTR _hProvider;
+
+        void initialize() @trusted
+        {
+            // https://msdn.microsoft.com/en-us/library/windows/desktop/aa379886(v=vs.85).aspx
+            // For performance reasons, we recommend that you set the pszContainer
+            // parameter to NULL and the dwFlags parameter to CRYPT_VERIFYCONTEXT
+            // in all situations where you do not require a persisted key.
+            // CRYPT_SILENT is intended for use with applications for which the UI cannot be displayed by the CSP.
+            if (!CryptAcquireContextW(&_hProvider, null, null, PROV_RSA_FULL, CRYPT_VERIFYCONTEXT | CRYPT_SILENT))
+            {
+                if (GetLastError() == NTE_BAD_KEYSET)
+                {
+                    // Attempt to create default container
+                    wenforce (CryptAcquireContextA(&_hProvider, null, null, PROV_RSA_FULL,
+                    CRYPT_NEWKEYSET | CRYPT_SILENT), "Could not acquire crypt context");
+                }
+                else
+                {
+                    wenforce (false, "Could not acquire crypt context");
+                }
+            }
+
+            popFront();
+        }
+
+    public:
+        enum bool isUniformRandom = true;
+        enum empty = false;
+
+        ulong front;
+
+        void popFront() @trusted nothrow @nogc
+        {
+            if (!CryptGenRandom(_hProvider, front.sizeof, cast(PBYTE) &front))
+                assert(0, "CryptGenRandom failed");
+        }
+
+        // TLS data is not destructed anyway right now...
+        version (none)
+        {
+            ~this()
+            {
+                CryptReleaseContext(_hProvider, 0);
+            }
+        }
+    }
+}
+
+version (SecureARC4Random)
+{
+    struct ARC4Random
+    {
+    private:
+        void initialize() @nogc @trusted nothrow
+        {
+            popFront();
+        }
+
+    public:
+        enum bool isUniformRandom = true;
+        enum empty = false;
+
+        ulong front;
+
+        void popFront() @safe nothrow @nogc
+        {
+            arc4random_buf(&front, front.sizeof);
```

Comment:
We can fetch more bytes here:

https://linux.die.net/man/3/arc4random_buf

Reply:
Any idea how much makes sense here? 64, 128 byte?

Reply:
We probably should benchmark this, but https://github.com/tcort/arc4random uses `256` by default and that's what my gut feeling suggested as well.